modules/misc/nixpkgs.nix

```diff
@@ -0,0 +1,29 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  nixpkgsModule = import <nixpkgs/nixos/modules/misc/nixpkgs.nix> {
+    inherit lib pkgs;
+    config.nixpkgs = removeAttrs config.nixpkgs [ "enable" ];
+  };
+
+in
+
+{
+  # We reuse the options from the NixOS module but we remove the
+  # `system` attribute since it is not particularly relevant for a
+  # user environment. We also add an `enable` option to allow the user
+  # to decide whether Home Manager should manage the Nixpkgs
+  # configuration.
+  options = {
+    nixpkgs =
+      removeAttrs nixpkgsModule.options.nixpkgs [ "system" ]
```

Comment:
`system` is useful when deploying a hm config to a different machine than the one it's built on, which will be relevant when the nixos module is integrated and used through nixops for example.

Reply:
Yeah, but in that case it should be taking the system option from the system configuration, not expect the user to set it.

So, instead of removing it completely I guess the best would be to mark it as internal.